taojieTan/assignments/mergeIntervals/lc57/lc57.py

```diff
@@ -2,17 +2,67 @@
 
 # Given a list of non-overlapping intervals sorted by their start time, insert a given interval at the correct position and merge all necessary intervals to produce a list that has only mutually exclusive intervals.
 
+# input: a list containing lists with non-overlapping intervals (sorted by start), and a list with start and end intervals  
+# output: a new list with non-overlapping intervals
+# edge cases: 1. an empty list and a new interval => insert the new interval to empty list and return it
+#             2. new interval doesn't overlap the list => insert it at the start/end of list and return it              
+#             3. new interval overlaps all intervals in the list => insert it into new list and return it
+
+# Let's call the start value of the new interval to be 'start', and the end value of the new interval to be 'end'.
+# Since the initial list is sorted and non-overlapping, we don't need to worry about if there is overlapping intervals in the list nor arranging them. 
+# We can loop through the list to first check: 1. if there are any intervals that are not overlapping the new interval and insert them into new list,
+# 2. then we will compare the end value of new interval with each start value of rest of intervals to find the overlapping part. 
+# We pick the smaller start value as the start of the overlapping part, and continue looping.
+# Once we found the overlapping interval, we merge it and insert it into new list.
+# 3. Then we can insert the rest of the intervals into the new list since they are non-overlapping. 
+# We will get the final result after the loops.
```

Comment:
Great use of high level overview! Small critique is maybe the readability is not best, try word wrapping at a certain point and if you need to make easier to read can introduce blank lines between thoughts